tests/connectors/dim/margo_test.py

```diff
@@ -198,3 +208,12 @@ def test_mocked_spawn_server() -> None:
 def test_wait_for_server_raises_error() -> None:
     with pytest.raises(ServerTimeoutError):
         wait_for_server('tcp', '127.0.0.1', 0, timeout=0)
+
+
+def test_provide_interface() -> None:
```

Comment:
Mock `wait_for_server` here so we don't cause side effects (also why the integration test runtime increased by 2-3 minutes).